test/AssemblySet_basic_test.py

```diff
@@ -1,54 +1,41 @@
 # -*- coding: utf-8 -*-
 import os
 import shutil
-import unittest
 from pprint import pprint
 from test import TEST_BASE_DIR
-from test.conftest import INFO_LENGTH, WS_NAME, test_config
+from test.base_class import BaseTestClass
+from test.conftest import INFO_LENGTH
 
-from installed_clients.AssemblyUtilClient import AssemblyUtil
 
-
-class SetAPITest(unittest.TestCase):
+class AssemblySetAPITest(BaseTestClass):
     @classmethod
-    def setUpClass(cls):
-        props = test_config()
-        for prop in ["cfg", "ctx", "serviceImpl", "wsClient", "wsName", "wsURL"]:
-            setattr(cls, prop, props[prop])
+    def prepare_data(cls: BaseTestClass) -> None:
```

Comment:
All the test classes have the same structure -- create the class and set up a workspace for testing; create fixtures for this class; tear down the class afterwards. I have created a base test class that does the set up and tear down and calls a `prepare_data` method where test fixtures are generated. Thus any child classes just need to implement the `prepare_data` method and everything else is handled by the base class.